pkg/blacklist/analyzer.go

```diff
@@ -155,54 +154,88 @@ func hasBlacklistedSrcQuery(chunk int, ip string, entry uconnRes, newFlag bool)
 
 		query["$push"] = bson.M{
 			"dat": bson.M{
-				"bl":             ip,
+				"bl":             blacklistedSrc,
 				"bl_in_count":    1,
-				"bl_total_bytes": entry.TotalBytes,
-				"bl_conn_count":  entry.Connections,
+				"bl_total_bytes": dstConnData.TotalBytes,
+				"bl_conn_count":  dstConnData.Connections,
 				"cid":            chunk,
 			}}
 
 		// create selector for output
 		output.query = query
-		output.selector = bson.M{"ip": entry.Host}
+		output.selector = dstConnData.Host.BSONKey()
 
 	} else {
 
 		query["$set"] = bson.M{
-			"dat.$.bl_conn_count":  entry.Connections,
-			"dat.$.bl_total_bytes": entry.TotalBytes,
+			"dat.$.bl_conn_count":  dstConnData.Connections,
+			"dat.$.bl_total_bytes": dstConnData.TotalBytes,
 			"dat.$.bl_in_count":    1,
 			"dat.$.cid":            chunk,
 		}
 
 		// create selector for output
-		output.query = query
-		output.selector = bson.M{"ip": entry.Host, "dat.bl": ip}
+		output.selector = dstConnData.Host.BSONKey()
+		output.selector["dat.bl"] = blacklistedSrc.BSONKey()
 	}
 
 	return output
 }
 
-// getBlaclistedIPConnections
-func (a *analyzer) getBlacklistedIPConnections(ip string, field1 string, field2 string) []uconnRes {
+//getUniqueConnsforBLDestination returns the IP addresses that contacted a given blacklisted IP along with the number
+//of connections and bytes sent
+func (a *analyzer) getUniqueConnsforBLDestination(blDestinationIP data.UniqueIP) []connectionPeer {
 	ssn := a.db.Session.Copy()
 	defer ssn.Close()
 
-	var blIPs []uconnRes
+	var blIPs []connectionPeer
 
 	blIPQuery := []bson.M{
-		bson.M{"$match": bson.M{field1: ip}},
+		bson.M{"$match": bson.M{
+			"dst":              blDestinationIP.IP,
+			"dst_network_uuid": blDestinationIP.NetworkUUID,
+		}},
 		bson.M{"$unwind": "$dat"},
 		bson.M{"$group": bson.M{
-			"_id":            "$" + field2,
+			"_id": bson.M{
+				"ip":           "$src",
+				"network_uuid": "$src_network_uuid",
+			},
 			"bl_conn_count":  bson.M{"$sum": "$dat.count"},
-			"bl_in_count":    bson.M{"$sum": 1},
 			"bl_total_bytes": bson.M{"$sum": "$dat.tbytes"},
 		}},
 	}
 
 	_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.UniqueConnTable).Pipe(blIPQuery).AllowDiskUse().All(&blIPs)
 
 	return blIPs
+}
+
+//getUniqueConnsforBLSource returns the IP addresses that a given blacklisted IP contacted along with the number
+//of connections and bytes sent
+func (a *analyzer) getUniqueConnsforBLSource(blSourceIP data.UniqueIP) []connectionPeer {
+	ssn := a.db.Session.Copy()
+	defer ssn.Close()
+
+	var blIPs []connectionPeer
 
+	blIPQuery := []bson.M{
+		bson.M{"$match": bson.M{
```

Comment:
@Zalgo2462

Reply:
Unfortunately, no, because we need to match on "src"/ "src_network_uuid" rather than on "ip"/ "network_uuid". 

The UniqueIPPair translates down to these fields but also includes "dst"/"dst_network_uuid".

One could imagine adding `AsSrcBSONKey` and `AsDstBSONKey` methods, but I didn't think of that while writing the code initially.